examples/ffi/exporter.cpp

```diff
@@ -40,9 +40,12 @@ int main(int argc, char *argv[]) {
 
   const ddog_prof_Slice_ValueType sample_types = {&wall_time, 1};
   const ddog_prof_Period period = {wall_time, 60};
-  ddog_prof_Profile profile_data{
-      ddog_prof_Profile_new(sample_types, &period, nullptr)};
-  std::unique_ptr<ddog_prof_Profile, Deleter> profile{&profile_data};
+  ddog_prof_Profile_NewResult profile_new_result =
+      ddog_prof_Profile_new(sample_types, &period, nullptr);
+  if (profile_new_result.tag != DDOG_PROF_PROFILE_NEW_RESULT_OK) {
+    print_error("Failed to make new profile: ", profile_new_result.err);
+  }
```

Comment:
Are we missing the `ddog_Error_drop` here? I'm thinking it's relevant to have it in the example, as people may be inspired to copy/paste from here, and this way they won't forget the cleanup.

Also, like some of the other error checks, I think we should probably terminate the example if there's an error, since right now we're continuing with a weird broken state.

Reply:
Is there any way we could make this clearer ? 
Especially as we have made this mistake many times.  @danielsn would you have ideas ? Maybe even naming ?

Reply:
Yes, we are missing some error stuff here. I wrote this part kind of hastily. Sorry about that. I'll fix it later.

Reply:
For C++, could we wrap some of this stuff in smart pointers, guaranteeing that the drops happen.  I wonder if it would be worth having a seperate C and C++ API to give more protection to the C++ side

Reply:
I will leave that up to the C++ users (maybe native + .NET?)